### PR TITLE
feat: add equality and hashCode overrides

### PR DESCRIPTION
**Pull request by td-famedly**
helper function overrides

### CODE REVIEW

File: bin/dart_openapi_codegen.dart [hunk 1]
```diff
@@ -16,7 +16,9 @@ String className(String s) => fixup(s
     .map((x) => x.capitalized)
     .join(''));
 
-String variableName(String s) => className(s).uncapitalized;
+// TODO: maybe make the keyword escape more robust later
+String variableName(String s) =>
+    s == 'override' ? 'overrideField' : className(s).uncapitalized;
```
Comment (nico-famedly) [inline bin/dart_openapi_codegen.dart:21]: Since when is override a keyword?

Comment (td-famedly) [reply bin/dart_openapi_codegen.dart:21]: well you can't have a field name called override :D

Comment (td-famedly) [reply bin/dart_openapi_codegen.dart:21]: it could just be my linter actually brb

Comment (td-famedly) [reply bin/dart_openapi_codegen.dart:21]: ok no, you can't have @override and a field name called override in the same class

```
• lib/matrix_api_lite/generated/model.dart:2875:3
Annotation must be either a const reference or constructor invalid_annotation
          ```

**Review by nico-famedly — APPROVED** [review]

File: bin/dart_openapi_codegen.dart [hunk 2]
```diff
@@ -921,7 +929,7 @@ void numberConflicts(List<Operation> operations) {
 }
 
 String generateModel(List<Operation> operations) {
-  return "import 'internal.dart';\nimport 'package:enhanced_enum/enhanced_enum.dart';\npart 'model.g.dart';\nclass _NameSource { final String source; const _NameSource(this.source); }\n\n" +
+  return "import 'dart:core' as dart; import 'dart:core'; import 'internal.dart';\nimport 'package:enhanced_enum/enhanced_enum.dart';\npart 'model.g.dart';\nclass _NameSource { final String source; const _NameSource(this.source); }\n\n" +
```
[the remark this answers is not in the document]
Comment (td-famedly) [reply bin/dart_openapi_codegen.dart:932]: yes but those are thrown out of whack my the formatter changing order of the imports, i'll just let people search where the @dart namespace is used

**Pull request merged**